### PR TITLE
some additional helper functions

### PR DESCRIPTION
**Pull request by olakiril**

### CODE REVIEW

File: lib/+ne7/+mat/getParams.m [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+% function params = getParams(params,vararginput)
+%
+% Gives the ability to pass the param as a structure to a function
+%
+% 2009-02-24 MF
+
+
+if ~isempty(vararginput)
+
+    char = zeros(1,length(vararginput));
```
Comment (eywalker) [inline lib/+ne7/+mat/getParams.m]: Since both `char` and `struc` are built-in Matlab functions, would you mind changing them to something else?

**Review by eywalker — CHANGES_REQUESTED** [review]
Avoid use of built-in Matlab functions as variable names.

**Comment by olakiril** [comment]
Edited on my mobile so adding 2 was the easiest thing to do...

File: lib/+ne7/+ip/imgaussian.m [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+%
+% note, compile the code with: mex imgaussian.c -v
+%
+% example,
+%   I = im2double(imread('peppers.png'));
+%   figure, imshow(imgaussian(I,10));
+% 
+% Function is written by D.Kroon University of Twente (September 2009)
+
+if(~exist('siz','var')), siz=sigma*6; end
```
Comment (dimitri-yatsenko) [inline lib/+ne7/+ip/imgaussian.m:27]: 6 sigma is excessive.  2.5 sigma is usually more than enough.

File: lib/+ne7/+ip/imgaussian.m [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+if(sigma>0)
+    % Make 1D Gaussian kernel
+    x=-ceil(siz/2):ceil(siz/2);
+    H = exp(-(x.^2/(2*sigma^2)));
+    H = H/sum(H(:));
+
+    % Filter each dimension with the 1D Gaussian kernels\
+    if(ndims(I)==1)
```
Comment (dimitri-yatsenko) [inline lib/+ne7/+ip/imgaussian.m:36]: The conditional blocks below contain repeated blocks and can be replaced by a single for loop.

File: lib/+ne7/+ip/imgaussian.m [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        I=imfilter(I,H, 'same' ,'replicate');
+    elseif(ndims(I)==2)
+        Hx=reshape(H,[length(H) 1]);
+        Hy=reshape(H,[1 length(H)]);
+        I=imfilter(imfilter(I,Hx, 'same' ,'replicate'),Hy, 'same' ,'replicate');
+    elseif(ndims(I)==3)
+        if(size(I,3)<4) % Detect if 3D or color image
```
Comment (dimitri-yatsenko) [inline lib/+ne7/+ip/imgaussian.m:43]: How do you know if it's a color image? If its 3rd dimension is 2 or 3?  A more reliable test or option should be used.

File: lib/+ne7/+ip/imgaussian.m [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+            Hx=reshape(H,[length(H) 1]);
+            Hy=reshape(H,[1 length(H)]);
+            for k=1:size(I,3)
```
Comment (dimitri-yatsenko) [inline lib/+ne7/+ip/imgaussian.m:46]: this loop is unnecessary since imfilter already filters each slice in the 3rd dimension.

**Review by dimitri-yatsenko — APPROVED** [review]
I approve but suggest improvements.

File: lib/+ne7/+mat/getParams.m [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    struc = zeros(1,length(vararginput));
+    
+    % find single and multiple params
+    for i = 1:length(vararginput)
```
Comment (dimitri-yatsenko) [inline lib/+ne7/+mat/getParams.m]: c = cellfun(@ischar, vararginput)
s = cellfun(@isstract, vararginput)

File: lib/+ne7/+mat/getParams.m [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+        a = fields(vararginput{struc});
+        for i = 1:size(a,1)
+            params.(cell2mat(a(i)))= vararginput{struc}.(cell2mat(a(i)));
+        end
+    end
+
+    % assign single params
+    for i = 1:length(char)-1
+        if char(i)==1
+            char(i+1)=0;
```
Comment (dimitri-yatsenko) [inline lib/+ne7/+mat/getParams.m]: so if something is not preceded by a string, it is simply skipped with no error message?

File: lib/+ne7/+mat/getParams.m [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    struc = zeros(1,length(vararginput));
+    
+    % find single and multiple params
+    for i = 1:length(vararginput)
+        char(i) = ischar(vararginput{i});
+        struc(i) = isstruct(vararginput{i});
+    end
+
+    % assign params in structure
+    if ~sum(struc)==0
```
Comment (dimitri-yatsenko) [inline lib/+ne7/+mat/getParams.m]: `sum(struc)~=0` or `sum(struc)>0`

**Comment by eywalker** [comment]
I agree with @dimitri-yatsenko in that there are a number of things that can be improved, but I'm approving this now for us to edit in the future.

**Review by eywalker — APPROVED** [review]

**Pull request merged**